File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        m = {}
        for i in range(len(nums)):          
                m[nums[i]] = 1 + m.get(nums[i],0)

        for key,value in m.items():
            if value > (n//2):
                return key

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
    
       if(nums.length == 0){
        return 0;
       }

       HashSet<Integer> temp = new HashSet<>();
       for(int n: nums){
        temp.add(n);
       }

        int maxLength = 0;
       for(int num : temp){
        if(temp.contains(num - 1)){
            continue;
        }
        int currNum = num;
        int currLength = 1;

        while(temp.contains(currNum + 1)){
            currNum++;
            currLength++;
        }
        maxLength = Math.max(maxLength,currLength);
       }

       return maxLength;

    }
}
